diff --git a/tests/test_uninstall.py b/tests/test_uninstall.py
index 9294251..f6d7bf9 100644
--- a/tests/test_uninstall.py
+++ b/tests/test_uninstall.py
@@ -297,11 +297,14 @@ def test_uninstallation_menuinst(
     (True, False),
     ids=("shared pkgs", "remove pkgs"),
 )
+@pytest.mark.skipif(not ON_CI, reason="CI only - may remove shared caches")
 def test_uninstallation_remove_caches(
     mock_system_paths: dict[str, Path],
     tmp_env: TmpEnvFixture,
     shared_pkgs: bool,
 ):
+    if "CONDA_PKGS_DIRS" in os.environ:
+        pytest.skip("Test will fail with CONDA_PKGS_DIRS set.")
     # Set up notices
     if ON_WIN:
         try:
