diff --git a/menuinst/__init__.py b/menuinst/__init__.py
index 1dad2af..8d402c2 100644
--- a/menuinst/__init__.py
+++ b/menuinst/__init__.py
@@ -56,6 +56,7 @@ def install(path, remove=False, prefix=sys.prefix, recursing=False, root_prefix=
     # `sys.prefix`, therefore we need to specify it  
     """
     # this root_prefix is intentional.  We want to reflect the state of the root installation.
+    root_prefix = prefix
     if sys.platform == 'win32' and not exists(join(root_prefix, '.nonadmin')):
         if isUserAdmin():
             _install(path, remove, prefix, mode='system', root_prefix=root_prefix)
