diff --git a/menuinst/win32.py b/menuinst/win32.py
index 6aabd93..370909e 100644
--- a/menuinst/win32.py
+++ b/menuinst/win32.py
@@ -171,7 +171,7 @@ def substitute_env_variables(text, dir):
     for a, b in (
         (u'${PREFIX}', env_prefix),
         (u'${ROOT_PREFIX}', root_prefix),
-        (u'${DISTRIBUTION_NAME}', os.path.split(root_prefix)[-1].capitalize()),
+        (u'${DISTRIBUTION_NAME}', os.path.split(root_prefix)[-1]),
         (u'${PYTHON_SCRIPTS}',
           os.path.normpath(join(env_prefix, u'Scripts')).replace(u"\\", u"/")),
         (u'${MENU_DIR}', join(env_prefix, u'Menu')),
