From d0139d816cd8a4773a5a07c4109bbf74917794f3 Mon Sep 17 00:00:00 2001
From: Jaida Rice <jaidarice@gmail.com>
Date: Fri, 8 Aug 2025 15:26:11 -0700
Subject: [PATCH] Unset `CONDARC` to detect .condarc file of conda-standalone

---
 tests/test_main.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/test_main.py b/tests/test_main.py
index effc743..ba0d7c4 100644
--- a/tests/test_main.py
+++ b/tests/test_main.py
@@ -70,6 +70,8 @@ def test_constructor_missing_arguments(args: list[str]):
 
 @pytest.mark.parametrize("search_paths", ("all_rcs", "--no-rc", "env_var"))
 def test_conda_standalone_config(search_paths, tmp_path, monkeypatch):
+    # Unset `CONDARC` to detect .condarc file of conda-standalone
+    monkeypatch.delenv("CONDARC", raising=False)
     variant = os.environ.get("PYINSTALLER_BUILD_VARIANT", "single-binary")
     expected_configs = {}
     yaml = YAML()
-- 
2.32.1 (Apple Git-133)

