From 7e2b47b2073f302d01dc2b423549f03c21f52c9e Mon Sep 17 00:00:00 2001
From: Marco Esters <mesters@anaconda.com>
Date: Thu, 31 Jul 2025 06:18:47 -0700
Subject: [PATCH 3/3] Restrict search paths

---
 conda/base/constants.py | 68 ++++++++++++++++++++++-------------------
 1 file changed, 37 insertions(+), 31 deletions(-)

diff --git a/conda/base/constants.py b/conda/base/constants.py
index 5b42cf84a..94955fd53 100644
--- a/conda/base/constants.py
+++ b/conda/base/constants.py
@@ -12,6 +12,7 @@ from __future__ import annotations

 import struct
 from enum import Enum, EnumMeta
+from os import environ
 from os.path import join
 from typing import TYPE_CHECKING

@@ -35,42 +36,47 @@ APP_NAME: Final = "conda"

 SEARCH_PATH: tuple[str, ...]

-if on_win:  # pragma: no cover
+if "CONDA_RESTRICT_RC_SEARCH_PATH" in environ:
     SEARCH_PATH = (
-        "C:/ProgramData/conda/.condarc",
-        "C:/ProgramData/conda/condarc",
-        "C:/ProgramData/conda/condarc.d",
+        "$CONDARC",
     )
 else:
-    SEARCH_PATH = (
-        "/etc/conda/.condarc",
-        "/etc/conda/condarc",
-        "/etc/conda/condarc.d/",
-        "/var/lib/conda/.condarc",
-        "/var/lib/conda/condarc",
-        "/var/lib/conda/condarc.d/",
+    if on_win:  # pragma: no cover
+        SEARCH_PATH = (
+            "C:/ProgramData/conda/.condarc",
+            "C:/ProgramData/conda/condarc",
+            "C:/ProgramData/conda/condarc.d",
+        )
+    else:
+        SEARCH_PATH = (
+            "/etc/conda/.condarc",
+            "/etc/conda/condarc",
+            "/etc/conda/condarc.d/",
+            "/var/lib/conda/.condarc",
+            "/var/lib/conda/condarc",
+            "/var/lib/conda/condarc.d/",
+        )
+
+    SEARCH_PATH += (
+        "$CONDA_ROOT/.condarc",
+        "$CONDA_ROOT/condarc",
+        "$CONDA_ROOT/condarc.d/",
+        "$XDG_CONFIG_HOME/conda/.condarc",
+        "$XDG_CONFIG_HOME/conda/condarc",
+        "$XDG_CONFIG_HOME/conda/condarc.d/",
+        "~/.config/conda/.condarc",
+        "~/.config/conda/condarc",
+        "~/.config/conda/condarc.d/",
+        "~/.conda/.condarc",
+        "~/.conda/condarc",
+        "~/.conda/condarc.d/",
+        "~/.condarc",
+        "$CONDA_PREFIX/.condarc",
+        "$CONDA_PREFIX/condarc",
+        "$CONDA_PREFIX/condarc.d/",
+        "$CONDARC",
     )

-SEARCH_PATH += (
-    "$CONDA_ROOT/.condarc",
-    "$CONDA_ROOT/condarc",
-    "$CONDA_ROOT/condarc.d/",
-    "$XDG_CONFIG_HOME/conda/.condarc",
-    "$XDG_CONFIG_HOME/conda/condarc",
-    "$XDG_CONFIG_HOME/conda/condarc.d/",
-    "~/.config/conda/.condarc",
-    "~/.config/conda/condarc",
-    "~/.config/conda/condarc.d/",
-    "~/.conda/.condarc",
-    "~/.conda/condarc",
-    "~/.conda/condarc.d/",
-    "~/.condarc",
-    "$CONDA_PREFIX/.condarc",
-    "$CONDA_PREFIX/condarc",
-    "$CONDA_PREFIX/condarc.d/",
-    "$CONDARC",
-)
-
 DEFAULT_CHANNEL_ALIAS: Final = "https://conda.anaconda.org"
 CONDA_HOMEPAGE_URL: Final = "https://conda.io"
 ERROR_UPLOAD_URL: Final = "https://conda.io/conda-post/unexpected-error"
--
2.32.1 (Apple Git-133)

