From da161195d96c6b5a755409e74e388b17d769e110 Mon Sep 17 00:00:00 2001
From: fwaters <fwaters@anaconda.com>
Date: Mon, 7 Oct 2019 11:48:31 -0700
Subject: [PATCH] fix entry_point name for standalone

---
 conda/core/path_actions.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/conda/core/path_actions.py b/conda/core/path_actions.py
index 9567764b3..01a1316d1 100644
--- a/conda/core/path_actions.py
+++ b/conda/core/path_actions.py
@@ -258,8 +258,14 @@ class LinkPathAction(CreateInPrefixPathAction):
     def create_python_entry_point_windows_exe_action(cls, transaction_context, package_info,
                                                      target_prefix, requested_link_type,
                                                      entry_point_def):
-        source_directory = context.conda_prefix
-        source_short_path = 'Scripts/conda.exe'
+        import os
+
+        if lexists(sys.argv[0]):
+            source_directory = os.path.dirname(sys.argv[0])
+            source_short_path = os.path.basename(sys.argv[0])
+        else:
+            source_directory = context.conda_prefix
+            source_short_path = 'Scripts/conda.exe'
         command, _, _ = parse_entry_point_def(entry_point_def)
         target_short_path = "Scripts/%s.exe" % command
         source_path_data = PathDataV1(
--
2.17.0.windows.1
