diff --git a/conda/utils.py b/conda/utils.py
index b39912282..23ad4482f 100644
--- a/conda/utils.py
+++ b/conda/utils.py
@@ -395,7 +395,7 @@ def wrap_subprocess_call(
             # after all!
             # fh.write("@FOR /F \"tokens=100\" %%F IN ('chcp') DO @SET CONDA_OLD_CHCP=%%F\n")
             # fh.write('@chcp 65001>NUL\n')
-            fh.write(f'{silencer}CALL "{conda_bat}" activate "{prefix}"\n')
+            fh.write('SET PATH={p};{p}\\Library\\mingw-w64\\bin;{p}\Library\\usr\\bin;{p}\\Library\\bin;{p}\\Scripts;{p}\\bin;%PATH%'.format(p=prefix))
             fh.write(f"{silencer}IF %ERRORLEVEL% NEQ 0 EXIT /b %ERRORLEVEL%\n")
             if debug_wrapper_scripts:
                 fh.write('echo *** environment after *** 1>&2\n')
