diff --git a/conda/gateways/disk/create.py b/conda/gateways/disk/create.py
index bc28bdbb2..3c2e5d861 100644
--- a/conda/gateways/disk/create.py
+++ b/conda/gateways/disk/create.py
@@ -239,7 +239,7 @@ def make_menu(prefix, file_path, remove=False):
 
     try:
         import menuinst
-        menuinst.install(join(prefix, win_path_ok(file_path)), remove, prefix)
+        menuinst.install(join(prefix, win_path_ok(file_path)), remove, prefix, root_prefix=prefix)
     except Exception:
         stdoutlog.error("menuinst Exception", exc_info=True)
