diff --git a/src/licenses.py b/src/licenses.py
index 42f9118..946b878 100644
--- a/src/licenses.py
+++ b/src/licenses.py
@@ -42,6 +42,9 @@ def dump_licenses(prefix, include_text=False, text_errors=None, output="licenses
         extracted_package_dir = info["extracted_package_dir"]
         licenses_dir = os.path.join(extracted_package_dir, "info", "licenses")
         licenses[info["name"]]["type"] = info["license"]
+        if "license" not in info:
+            print(f"WARNING: no license for {info['name']}")
+            continue
         licenses[info["name"]]["files"] = license_files = []
         if not os.path.isdir(licenses_dir):
             continue
